plugins/modules/packaging/os/pacman_key.py

```diff
@@ -0,0 +1,315 @@
+        type: bool
+        default: no
+    keyring:
+        description:
+            - The full path to the keyring folder on the remote server.
+            - If not specified, module will use pacman's default (C(/etc/pacman.d/gnupg)).
+            - Useful if the remote system requires an alternative gnupg directory.
+        type: path
+        default: /etc/pacman.d/gnupg
+    state:
+        description:
+            - Ensures that the key is present (added) or absent (revoked).
+        default: present
+        choices: [ absent, present ]
+        type: str
+'''
+
+EXAMPLES = '''
+- name: Import a key via local file
+  community.general.pacman_key:
+    data: "{{ lookup('file', 'keyfile.asc') }}"
+    state: present
+
+- name: Import a key via remote file
+  community.general.pacman_key:
+    file: /tmp/keyfile.asc
+    state: present
+
+- name: Import a key via url
+  community.general.pacman_key:
+    id: 01234567890ABCDE01234567890ABCDE12345678
+    url: https://domain.tld/keys/keyfile.asc
+    state: present
+
+- name: Import a key via keyserver
+  community.general.pacman_key:
+    id: 01234567890ABCDE01234567890ABCDE12345678
+    keyserver: keyserver.domain.tld
+
+- name: Import a key into an alternative keyring
+  community.general.pacman_key:
+    id: 01234567890ABCDE01234567890ABCDE12345678
+    file: /tmp/keyfile.asc
+    keyring: /etc/pacman.d/gnupg-alternative
+
+- name: Remove a key from the keyring
+  community.general.pacman_key:
+    id: 01234567890ABCDE01234567890ABCDE12345678
+    state: absent
+'''
+
+RETURN = r''' # '''
+
+import os.path
+import tempfile
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils._text import to_native
+
+
+class PacmanKey(object):
+    def __init__(self, module):
+        self.module = module
+        # obtain binary paths for gpg & pacman-key
+        self.gpg = module.get_bin_path('gpg', required=True)
+        self.pacman_key = module.get_bin_path('pacman-key', required=True)
+
+        # obtain module parameters
+        keyid = module.params['id']
+        url = module.params['url']
+        data = module.params['data']
+        file = module.params['file']
+        keyserver = module.params['keyserver']
+        verify = module.params['verify']
+        force_update = module.params['force_update']
+        keyring = module.params['keyring']
+        state = module.params['state']
+        self.keylength = 40
+
+        # sanitise key ID & check if key exists in the keyring
+        keyid = self.sanitise_keyid(keyid)
+        key_present = self.key_in_keyring(keyring, keyid)
+
+        # check mode
+        if module.check_mode:
+            if state == "present":
+                if (key_present and force_update) or not key_present:
+                    module.exit_json(changed=True)
+                module.exit_json(changed=False)
```

Comment:
Nit picking alert!
```suggestion
                changed = (key_present and force_update) or not key_present
                module.exit_json(changed=changed)
```
Yours truly personal preferences lean towards less conditional statements. Feel free to ignore as it is just a matter of styles.